src/modules/Telemetry/Sensor/AHT10.cpp

```diff
@@ -0,0 +1,35 @@
+#include "AHT10.h"
+#include "../mesh/generated/meshtastic/telemetry.pb.h"
+#include "TelemetrySensor.h"
+#include "configuration.h"
+#include <Adafruit_AHTX0.h>
+#include <typeinfo>
+
+AHT10Sensor::AHT10Sensor() : TelemetrySensor(meshtastic_TelemetrySensorType_AHT10, "AHT10") {}
+
+int32_t AHT10Sensor::runOnce()
+{
+    LOG_INFO("Init sensor: %s\n", sensorName);
+    if (!hasSensor()) {
+        return DEFAULT_SENSOR_MINIMUM_WAIT_TIME_BETWEEN_READS;
+    }
+    aht10 = Adafruit_AHTX0();
+    status = aht10.begin(nodeTelemetrySensorsMap[sensorType].second, 0, nodeTelemetrySensorsMap[sensorType].first);
+
+    return initI2CSensor();
+}
+
+void AHT10Sensor::setup() {}
+
+bool AHT10Sensor::getMetrics(meshtastic_Telemetry *measurement)
+{
+    LOG_DEBUG("AHT10Sensor::getMetrics\n");
+
+    sensors_event_t humidity, temp;
+    aht10.getEvent(&humidity, &temp);
+
+    measurement->variant.environment_metrics.temperature = temp.temperature;
+    measurement->variant.environment_metrics.relative_humidity = humidity.relative_humidity;
```

Comment:
Is this wise? you quote the sensor is used in addition to BME280. This will overwrite the temperature value. Maybe you can set this field only if the BME280 was NOT detected.

Reply:
Both sensors provide the temperature, so taking it from one or the other is the same thing.

Reply:
The problem is the measurements vary and which chi is used is not deteministic. I'd prefer the Bosch chip over the AHT